src/dialect/baseDialect.ts

```diff
@@ -155,21 +159,37 @@ export abstract class SQLDialect {
     return `CHAR_LENGTH(${a})`;
   }
 
-  public abstract timeFloorExpression(operand: string, duration: Duration, timezone: Timezone): string;
+  public abstract timeFloorExpression(
+    operand: string,
+    duration: Duration,
+    timezone: Timezone,
+  ): string;
 
-  public abstract timeBucketExpression(operand: string, duration: Duration, timezone: Timezone): string;
+  public abstract timeBucketExpression(
+    operand: string,
+    duration: Duration,
+    timezone: Timezone,
+  ): string;
 
   public abstract timePartExpression(operand: string, part: string, timezone: Timezone): string;
 
-  public abstract timeShiftExpression(operand: string, duration: Duration, step: int, timezone: Timezone): string;
+  public abstract timeShiftExpression(
+    operand: string,
+    duration: Duration,
+    step: int,
+    timezone: Timezone,
+  ): string;
 
   public abstract extractExpression(operand: string, regexp: string): string;
 
   public abstract indexOfExpression(str: string, substr: string): string;
 
+  public quantileExpression(str: string, substr: string): string {
```

Comment:
Is this ever going to be implemented for other dialects? If not it would make me slightly happier if this only existed on `DruidDialect` and we had `_getSQLChainableUnaryHelper` throw when it's incompatible with the dialect

Reply:
we effectively do that through inheritance. Base dialect has:

![image](https://user-images.githubusercontent.com/177816/86065132-b619fe80-ba23-11ea-954c-9619aa3fde56.png)